ThunkLibs/include/common/Host.h

```diff
@@ -192,3 +193,14 @@ void FinalizeHostTrampolineForGuestFunction(F* PreallocatedTrampolineForGuestFun
       (FEXCore::HostToGuestTrampolinePtr*)PreallocatedTrampolineForGuestFunction,
       (void*)&CallbackUnpack<F>::CallGuestPtr);
 }
+
+// In the case of the thunk host_loader being the default, FEX need to use dlsym with RTLD_DEFAULT.
+// If FEX queried the symbol object directly then it wouldn't follow symbol overriding rules.
+//
+// Common usecase is LD_PRELOAD with a library that defines some symbols.
+// And then programs and libraries will pick up the preloaded symbols.
+// ex: MangoHud overrides GLX and EGL symbols.
+inline
+void *dlsym_default(void *__restrict handle, const char *__restrict symbol) {
```

Comment:
I don't see this mattering here, and I'd prefer to keep utility code simple and readable.

If `__restrict` were to make a difference anywhere, we'd get it anyway since `dlsym_default` is trivially inlined and doesn't override `dlsym`'s `__restrict` qualifiers.

Reply:
Sure thing, removed